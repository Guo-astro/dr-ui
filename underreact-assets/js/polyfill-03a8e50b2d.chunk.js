(window.webpackJsonp=window.webpackJsonp||[]).push([[3],{12:function(t,n,r){var e=r(50)("wks"),o=r(35),i=r(16).Symbol,u="function"==typeof i;(t.exports=function(t){return e[t]||(e[t]=u&&i[t]||(u?i:o)("Symbol."+t))}).store=e},135:function(t,n,r){t.exports=!r(26)&&!r(48)(function(){return 7!=Object.defineProperty(r(136)("div"),"a",{get:function(){return 7}}).a})},136:function(t,n,r){var e=r(33),o=r(16).document,i=e(o)&&e(o.createElement);t.exports=function(t){return i?o.createElement(t):{}}},137:function(t,n,r){var e=r(530);t.exports=function(t,n,r){if(e(t),void 0===n)return t;switch(r){case 1:return function(r){return t.call(n,r)};case 2:return function(r,e){return t.call(n,r,e)};case 3:return function(r,e,o){return t.call(n,r,e,o)}}return function(){return t.apply(n,arguments)}}},138:function(t,n,r){n.f=r(12)},139:function(t,n,r){var e=r(19),o=r(36),i=r(535)(!1),u=r(85)("IE_PROTO");t.exports=function(t,n){var r,c=o(t),f=0,a=[];for(r in c)r!=u&&e(c,r)&&a.push(r);for(;n.length>f;)e(c,r=n[f++])&&(~i(a,r)||a.push(r));return a}},140:function(t,n,r){var e=r(84),o=Math.min;t.exports=function(t){return t>0?o(e(t),9007199254740991):0}},141:function(t,n){n.f=Object.getOwnPropertySymbols},142:function(t,n,r){var e=r(32),o=r(538),i=r(86),u=r(85)("IE_PROTO"),c=function(){},f=function(){var t,n=r(136)("iframe"),e=i.length;for(n.style.display="none",r(539).appendChild(n),n.src="javascript:",(t=n.contentWindow.document).open(),t.write("<script>document.F=Object<\/script>"),t.close(),f=t.F;e--;)delete f.prototype[i[e]];return f()};t.exports=Object.create||function(t,n){var r;return null!==t?(c.prototype=e(t),r=new c,c.prototype=null,r[u]=t):r=f(),void 0===n?r:o(r,n)}},143:function(t,n,r){var e=r(139),o=r(86).concat("length","prototype");n.f=Object.getOwnPropertyNames||function(t){return e(t,o)}},144:function(t,n,r){var e=r(82),o=r(12)("toStringTag"),i="Arguments"==e(function(){return arguments}());t.exports=function(t){var n,r,u;return void 0===t?"Undefined":null===t?"Null":"string"==typeof(r=function(t,n){try{return t[n]}catch(r){}}(n=Object(t),o))?r:i?e(n):"Object"==(u=e(n))&&"function"==typeof n.callee?"Arguments":u}},145:function(t,n,r){"use strict";var e=r(559);function o(){}var i=null,u={};function c(t){if("object"!==typeof this)throw new TypeError("Promises must be constructed via new");if("function"!==typeof t)throw new TypeError("Promise constructor's argument is not a function");this._h=0,this._i=0,this._j=null,this._k=null,t!==o&&v(t,this)}function f(t,n){for(;3===t._i;)t=t._j;if(c._l&&c._l(t),0===t._i)return 0===t._h?(t._h=1,void(t._k=n)):1===t._h?(t._h=2,void(t._k=[t._k,n])):void t._k.push(n);!function(t,n){e(function(){var r=1===t._i?n.onFulfilled:n.onRejected;if(null!==r){var e=function(t,n){try{return t(n)}catch(r){return i=r,u}}(r,t._j);e===u?s(n.promise,i):a(n.promise,e)}else 1===t._i?a(n.promise,t._j):s(n.promise,t._j)})}(t,n)}function a(t,n){if(n===t)return s(t,new TypeError("A promise cannot be resolved with itself."));if(n&&("object"===typeof n||"function"===typeof n)){var r=function(t){try{return t.then}catch(n){return i=n,u}}(n);if(r===u)return s(t,i);if(r===t.then&&n instanceof c)return t._i=3,t._j=n,void l(t);if("function"===typeof r)return void v(r.bind(n),t)}t._i=1,t._j=n,l(t)}function s(t,n){t._i=2,t._j=n,c._m&&c._m(t,n),l(t)}function l(t){if(1===t._h&&(f(t,t._k),t._k=null),2===t._h){for(var n=0;n<t._k.length;n++)f(t,t._k[n]);t._k=null}}function p(t,n,r){this.onFulfilled="function"===typeof t?t:null,this.onRejected="function"===typeof n?n:null,this.promise=r}function v(t,n){var r=!1,e=function(t,n,r){try{t(n,r)}catch(e){return i=e,u}}(t,function(t){r||(r=!0,a(n,t))},function(t){r||(r=!0,s(n,t))});r||e!==u||(r=!0,s(n,i))}t.exports=c,c._l=null,c._m=null,c._n=o,c.prototype.then=function(t,n){if(this.constructor!==c)return function(t,n,r){return new t.constructor(function(e,i){var u=new c(o);u.then(e,i),f(t,new p(n,r,u))})}(this,t,n);var r=new c(o);return f(this,new p(t,n,r)),r}},16:function(t,n){var r=t.exports="undefined"!=typeof window&&window.Math==Math?window:"undefined"!=typeof self&&self.Math==Math?self:Function("return this")();"number"==typeof __g&&(__g=r)},19:function(t,n){var r={}.hasOwnProperty;t.exports=function(t,n){return r.call(t,n)}},20:function(t,n){var r=t.exports={version:"2.6.9"};"number"==typeof __e&&(__e=r)},21:function(t,n,r){var e=r(32),o=r(135),i=r(79),u=Object.defineProperty;n.f=r(26)?Object.defineProperty:function(t,n,r){if(e(t),n=i(n,!0),e(r),o)try{return u(t,n,r)}catch(c){}if("get"in r||"set"in r)throw TypeError("Accessors not supported!");return"value"in r&&(t[n]=r.value),t}},26:function(t,n,r){t.exports=!r(48)(function(){return 7!=Object.defineProperty({},"a",{get:function(){return 7}}).a})},31:function(t,n,r){var e=r(21),o=r(34);t.exports=r(26)?function(t,n,r){return e.f(t,n,o(1,r))}:function(t,n,r){return t[n]=r,t}},32:function(t,n,r){var e=r(33);t.exports=function(t){if(!e(t))throw TypeError(t+" is not an object!");return t}},33:function(t,n){t.exports=function(t){return"object"===typeof t?null!==t:"function"===typeof t}},34:function(t,n){t.exports=function(t,n){return{enumerable:!(1&t),configurable:!(2&t),writable:!(4&t),value:n}}},35:function(t,n){var r=0,e=Math.random();t.exports=function(t){return"Symbol(".concat(void 0===t?"":t,")_",(++r+e).toString(36))}},36:function(t,n,r){var e=r(534),o=r(83);t.exports=function(t){return e(o(t))}},48:function(t,n){t.exports=function(t){try{return!!t()}catch(n){return!0}}},49:function(t,n,r){var e=r(16),o=r(31),i=r(19),u=r(35)("src"),c=r(529),f=(""+c).split("toString");r(20).inspectSource=function(t){return c.call(t)},(t.exports=function(t,n,r,c){var a="function"==typeof r;a&&(i(r,"name")||o(r,"name",n)),t[n]!==r&&(a&&(i(r,u)||o(r,u,t[n]?""+t[n]:f.join(String(n)))),t===e?t[n]=r:c?t[n]?t[n]=r:o(t,n,r):(delete t[n],o(t,n,r)))})(Function.prototype,"toString",function(){return"function"==typeof this&&this[u]||c.call(this)})},50:function(t,n,r){var e=r(20),o=r(16),i=o["__core-js_shared__"]||(o["__core-js_shared__"]={});(t.exports=function(t,n){return i[t]||(i[t]=void 0!==n?n:{})})("versions",[]).push({version:e.version,mode:r(51)?"pure":"global",copyright:"\xa9 2019 Denis Pushkarev (zloirock.ru)"})},51:function(t,n){t.exports=!1},525:function(t,n,r){"use strict";r(526),r(543),r(556),r(557)},526:function(t,n,r){"use strict";r(527)},527:function(t,n,r){r(528),r(542),t.exports=r(20).Symbol},528:function(t,n,r){"use strict";var e=r(16),o=r(19),i=r(26),u=r(78),c=r(49),f=r(531).KEY,a=r(48),s=r(50),l=r(80),p=r(35),v=r(12),y=r(138),h=r(532),d=r(533),b=r(537),g=r(32),m=r(33),_=r(88),w=r(36),x=r(79),j=r(34),O=r(142),S=r(540),P=r(541),E=r(141),k=r(21),T=r(81),F=P.f,A=k.f,I=S.f,M=e.Symbol,N=e.JSON,R=N&&N.stringify,C=v("_hidden"),U=v("toPrimitive"),D={}.propertyIsEnumerable,J=s("symbol-registry"),W=s("symbols"),z=s("op-symbols"),K=Object.prototype,G="function"==typeof M&&!!E.f,H=e.QObject,q=!H||!H.prototype||!H.prototype.findChild,B=i&&a(function(){return 7!=O(A({},"a",{get:function(){return A(this,"a",{value:7}).a}})).a})?function(t,n,r){var e=F(K,n);e&&delete K[n],A(t,n,r),e&&t!==K&&A(K,n,e)}:A,Y=function(t){var n=W[t]=O(M.prototype);return n._k=t,n},L=G&&"symbol"==typeof M.iterator?function(t){return"symbol"==typeof t}:function(t){return t instanceof M},Q=function(t,n,r){return t===K&&Q(z,n,r),g(t),n=x(n,!0),g(r),o(W,n)?(r.enumerable?(o(t,C)&&t[C][n]&&(t[C][n]=!1),r=O(r,{enumerable:j(0,!1)})):(o(t,C)||A(t,C,j(1,{})),t[C][n]=!0),B(t,n,r)):A(t,n,r)},V=function(t,n){g(t);for(var r,e=d(n=w(n)),o=0,i=e.length;i>o;)Q(t,r=e[o++],n[r]);return t},X=function(t){var n=D.call(this,t=x(t,!0));return!(this===K&&o(W,t)&&!o(z,t))&&(!(n||!o(this,t)||!o(W,t)||o(this,C)&&this[C][t])||n)},Z=function(t,n){if(t=w(t),n=x(n,!0),t!==K||!o(W,n)||o(z,n)){var r=F(t,n);return!r||!o(W,n)||o(t,C)&&t[C][n]||(r.enumerable=!0),r}},$=function(t){for(var n,r=I(w(t)),e=[],i=0;r.length>i;)o(W,n=r[i++])||n==C||n==f||e.push(n);return e},tt=function(t){for(var n,r=t===K,e=I(r?z:w(t)),i=[],u=0;e.length>u;)!o(W,n=e[u++])||r&&!o(K,n)||i.push(W[n]);return i};G||(c((M=function(){if(this instanceof M)throw TypeError("Symbol is not a constructor!");var t=p(arguments.length>0?arguments[0]:void 0);return i&&q&&B(K,t,{configurable:!0,set:function n(r){this===K&&n.call(z,r),o(this,C)&&o(this[C],t)&&(this[C][t]=!1),B(this,t,j(1,r))}}),Y(t)}).prototype,"toString",function(){return this._k}),P.f=Z,k.f=Q,r(143).f=S.f=$,r(87).f=X,E.f=tt,i&&!r(51)&&c(K,"propertyIsEnumerable",X,!0),y.f=function(t){return Y(v(t))}),u(u.G+u.W+u.F*!G,{Symbol:M});for(var nt="hasInstance,isConcatSpreadable,iterator,match,replace,search,species,split,toPrimitive,toStringTag,unscopables".split(","),rt=0;nt.length>rt;)v(nt[rt++]);for(var et=T(v.store),ot=0;et.length>ot;)h(et[ot++]);u(u.S+u.F*!G,"Symbol",{for:function(t){return o(J,t+="")?J[t]:J[t]=M(t)},keyFor:function(t){if(!L(t))throw TypeError(t+" is not a symbol!");for(var n in J)if(J[n]===t)return n},useSetter:function(){q=!0},useSimple:function(){q=!1}}),u(u.S+u.F*!G,"Object",{create:function(t,n){return void 0===n?O(t):V(O(t),n)},defineProperty:Q,defineProperties:V,getOwnPropertyDescriptor:Z,getOwnPropertyNames:$,getOwnPropertySymbols:tt});var it=a(function(){E.f(1)});u(u.S+u.F*it,"Object",{getOwnPropertySymbols:function(t){return E.f(_(t))}}),N&&u(u.S+u.F*(!G||a(function(){var t=M();return"[null]"!=R([t])||"{}"!=R({a:t})||"{}"!=R(Object(t))})),"JSON",{stringify:function(t){for(var n,r,e=[t],o=1;arguments.length>o;)e.push(arguments[o++]);if(r=n=e[1],(m(n)||void 0!==t)&&!L(t))return b(n)||(n=function(t,n){if("function"==typeof r&&(n=r.call(this,t,n)),!L(n))return n}),e[1]=n,R.apply(N,e)}}),M.prototype[U]||r(31)(M.prototype,U,M.prototype.valueOf),l(M,"Symbol"),l(Math,"Math",!0),l(e.JSON,"JSON",!0)},529:function(t,n,r){t.exports=r(50)("native-function-to-string",Function.toString)},530:function(t,n){t.exports=function(t){if("function"!=typeof t)throw TypeError(t+" is not a function!");return t}},531:function(t,n,r){var e=r(35)("meta"),o=r(33),i=r(19),u=r(21).f,c=0,f=Object.isExtensible||function(){return!0},a=!r(48)(function(){return f(Object.preventExtensions({}))}),s=function(t){u(t,e,{value:{i:"O"+ ++c,w:{}}})},l=t.exports={KEY:e,NEED:!1,fastKey:function(t,n){if(!o(t))return"symbol"==typeof t?t:("string"==typeof t?"S":"P")+t;if(!i(t,e)){if(!f(t))return"F";if(!n)return"E";s(t)}return t[e].i},getWeak:function(t,n){if(!i(t,e)){if(!f(t))return!0;if(!n)return!1;s(t)}return t[e].w},onFreeze:function(t){return a&&l.NEED&&f(t)&&!i(t,e)&&s(t),t}}},532:function(t,n,r){var e=r(16),o=r(20),i=r(51),u=r(138),c=r(21).f;t.exports=function(t){var n=o.Symbol||(o.Symbol=i?{}:e.Symbol||{});"_"==t.charAt(0)||t in n||c(n,t,{value:u.f(t)})}},533:function(t,n,r){var e=r(81),o=r(141),i=r(87);t.exports=function(t){var n=e(t),r=o.f;if(r)for(var u,c=r(t),f=i.f,a=0;c.length>a;)f.call(t,u=c[a++])&&n.push(u);return n}},534:function(t,n,r){var e=r(82);t.exports=Object("z").propertyIsEnumerable(0)?Object:function(t){return"String"==e(t)?t.split(""):Object(t)}},535:function(t,n,r){var e=r(36),o=r(140),i=r(536);t.exports=function(t){return function(n,r,u){var c,f=e(n),a=o(f.length),s=i(u,a);if(t&&r!=r){for(;a>s;)if((c=f[s++])!=c)return!0}else for(;a>s;s++)if((t||s in f)&&f[s]===r)return t||s||0;return!t&&-1}}},536:function(t,n,r){var e=r(84),o=Math.max,i=Math.min;t.exports=function(t,n){return(t=e(t))<0?o(t+n,0):i(t,n)}},537:function(t,n,r){var e=r(82);t.exports=Array.isArray||function(t){return"Array"==e(t)}},538:function(t,n,r){var e=r(21),o=r(32),i=r(81);t.exports=r(26)?Object.defineProperties:function(t,n){o(t);for(var r,u=i(n),c=u.length,f=0;c>f;)e.f(t,r=u[f++],n[r]);return t}},539:function(t,n,r){var e=r(16).document;t.exports=e&&e.documentElement},540:function(t,n,r){var e=r(36),o=r(143).f,i={}.toString,u="object"==typeof window&&window&&Object.getOwnPropertyNames?Object.getOwnPropertyNames(window):[];t.exports.f=function(t){return u&&"[object Window]"==i.call(t)?function(t){try{return o(t)}catch(n){return u.slice()}}(t):o(e(t))}},541:function(t,n,r){var e=r(87),o=r(34),i=r(36),u=r(79),c=r(19),f=r(135),a=Object.getOwnPropertyDescriptor;n.f=r(26)?a:function(t,n){if(t=i(t),n=u(n,!0),f)try{return a(t,n)}catch(r){}if(c(t,n))return o(!e.f.call(t,n),t[n])}},542:function(t,n,r){"use strict";var e=r(144),o={};o[r(12)("toStringTag")]="z",o+""!="[object z]"&&r(49)(Object.prototype,"toString",function(){return"[object "+e(this)+"]"},!0)},543:function(t,n,r){"use strict";r(544)},544:function(t,n,r){r(545),r(550),t.exports=r(20).Array.from},545:function(t,n,r){"use strict";var e=r(546)(!0);r(547)(String,"String",function(t){this._t=String(t),this._i=0},function(){var t,n=this._t,r=this._i;return r>=n.length?{value:void 0,done:!0}:(t=e(n,r),this._i+=t.length,{value:t,done:!1})})},546:function(t,n,r){var e=r(84),o=r(83);t.exports=function(t){return function(n,r){var i,u,c=String(o(n)),f=e(r),a=c.length;return f<0||f>=a?t?"":void 0:(i=c.charCodeAt(f))<55296||i>56319||f+1===a||(u=c.charCodeAt(f+1))<56320||u>57343?t?c.charAt(f):i:t?c.slice(f,f+2):u-56320+(i-55296<<10)+65536}}},547:function(t,n,r){"use strict";var e=r(51),o=r(78),i=r(49),u=r(31),c=r(89),f=r(548),a=r(80),s=r(549),l=r(12)("iterator"),p=!([].keys&&"next"in[].keys()),v=function(){return this};t.exports=function(t,n,r,y,h,d,b){f(r,n,y);var g,m,_,w=function(t){if(!p&&t in S)return S[t];switch(t){case"keys":case"values":return function(){return new r(this,t)}}return function(){return new r(this,t)}},x=n+" Iterator",j="values"==h,O=!1,S=t.prototype,P=S[l]||S["@@iterator"]||h&&S[h],E=P||w(h),k=h?j?w("entries"):E:void 0,T="Array"==n&&S.entries||P;if(T&&(_=s(T.call(new t)))!==Object.prototype&&_.next&&(a(_,x,!0),e||"function"==typeof _[l]||u(_,l,v)),j&&P&&"values"!==P.name&&(O=!0,E=function(){return P.call(this)}),e&&!b||!p&&!O&&S[l]||u(S,l,E),c[n]=E,c[x]=v,h)if(g={values:j?E:w("values"),keys:d?E:w("keys"),entries:k},b)for(m in g)m in S||i(S,m,g[m]);else o(o.P+o.F*(p||O),n,g);return g}},548:function(t,n,r){"use strict";var e=r(142),o=r(34),i=r(80),u={};r(31)(u,r(12)("iterator"),function(){return this}),t.exports=function(t,n,r){t.prototype=e(u,{next:o(1,r)}),i(t,n+" Iterator")}},549:function(t,n,r){var e=r(19),o=r(88),i=r(85)("IE_PROTO"),u=Object.prototype;t.exports=Object.getPrototypeOf||function(t){return t=o(t),e(t,i)?t[i]:"function"==typeof t.constructor&&t instanceof t.constructor?t.constructor.prototype:t instanceof Object?u:null}},550:function(t,n,r){"use strict";var e=r(137),o=r(78),i=r(88),u=r(551),c=r(552),f=r(140),a=r(553),s=r(554);o(o.S+o.F*!r(555)(function(t){Array.from(t)}),"Array",{from:function(t){var n,r,o,l,p=i(t),v="function"==typeof this?this:Array,y=arguments.length,h=y>1?arguments[1]:void 0,d=void 0!==h,b=0,g=s(p);if(d&&(h=e(h,y>2?arguments[2]:void 0,2)),void 0==g||v==Array&&c(g))for(r=new v(n=f(p.length));n>b;b++)a(r,b,d?h(p[b],b):p[b]);else for(l=g.call(p),r=new v;!(o=l.next()).done;b++)a(r,b,d?u(l,h,[o.value,b],!0):o.value);return r.length=b,r}})},551:function(t,n,r){var e=r(32);t.exports=function(t,n,r,o){try{return o?n(e(r)[0],r[1]):n(r)}catch(u){var i=t.return;throw void 0!==i&&e(i.call(t)),u}}},552:function(t,n,r){var e=r(89),o=r(12)("iterator"),i=Array.prototype;t.exports=function(t){return void 0!==t&&(e.Array===t||i[o]===t)}},553:function(t,n,r){"use strict";var e=r(21),o=r(34);t.exports=function(t,n,r){n in t?e.f(t,n,o(0,r)):t[n]=r}},554:function(t,n,r){var e=r(144),o=r(12)("iterator"),i=r(89);t.exports=r(20).getIteratorMethod=function(t){if(void 0!=t)return t[o]||t["@@iterator"]||i[e(t)]}},555:function(t,n,r){var e=r(12)("iterator"),o=!1;try{var i=[7][e]();i.return=function(){o=!0},Array.from(i,function(){throw 2})}catch(u){}t.exports=function(t,n){if(!n&&!o)return!1;var r=!1;try{var i=[7],c=i[e]();c.next=function(){return{done:r=!0}},i[e]=function(){return c},t(i)}catch(u){}return r}},556:function(t,n,r){"use strict";Object.assign=r(60)},557:function(t,n,r){"use strict";"undefined"===typeof Promise&&(r(558).enable(),window.Promise=r(560))},558:function(t,n,r){"use strict";var e=r(145),o=[ReferenceError,TypeError,RangeError],i=!1;function u(){i=!1,e._l=null,e._m=null}function c(t,n){return n.some(function(n){return t instanceof n})}n.disable=u,n.enable=function(t){t=t||{},i&&u();i=!0;var n=0,r=0,f={};function a(n){(t.allRejections||c(f[n].error,t.whitelist||o))&&(f[n].displayId=r++,t.onUnhandled?(f[n].logged=!0,t.onUnhandled(f[n].displayId,f[n].error)):(f[n].logged=!0,function(t,n){console.warn("Possible Unhandled Promise Rejection (id: "+t+"):"),((n&&(n.stack||n))+"").split("\n").forEach(function(t){console.warn("  "+t)})}(f[n].displayId,f[n].error)))}e._l=function(n){2===n._i&&f[n._o]&&(f[n._o].logged?function(n){f[n].logged&&(t.onHandled?t.onHandled(f[n].displayId,f[n].error):f[n].onUnhandled||(console.warn("Promise Rejection Handled (id: "+f[n].displayId+"):"),console.warn('  This means you can ignore any previous messages of the form "Possible Unhandled Promise Rejection" with id '+f[n].displayId+".")))}(n._o):clearTimeout(f[n._o].timeout),delete f[n._o])},e._m=function(t,r){0===t._h&&(t._o=n++,f[t._o]={displayId:null,error:r,timeout:setTimeout(a.bind(null,t._o),c(r,o)?100:2e3),logged:!1})}}},559:function(t,n,r){"use strict";(function(n){function r(t){o.length||(e(),!0),o[o.length]=t}t.exports=r;var e,o=[],i=0,u=1024;function c(){for(;i<o.length;){var t=i;if(i+=1,o[t].call(),i>u){for(var n=0,r=o.length-i;n<r;n++)o[n]=o[n+i];o.length-=i,i=0}}o.length=0,i=0,!1}var f="undefined"!==typeof n?n:self,a=f.MutationObserver||f.WebKitMutationObserver;function s(t){return function(){var n=setTimeout(e,0),r=setInterval(e,50);function e(){clearTimeout(n),clearInterval(r),t()}}}e="function"===typeof a?function(t){var n=1,r=new a(t),e=document.createTextNode("");return r.observe(e,{characterData:!0}),function(){n=-n,e.data=n}}(c):s(c),r.requestFlush=e,r.makeRequestCallFromTimer=s}).call(this,r(22))},560:function(t,n,r){"use strict";var e=r(145);t.exports=e;var o=s(!0),i=s(!1),u=s(null),c=s(void 0),f=s(0),a=s("");function s(t){var n=new e(e._n);return n._i=1,n._j=t,n}e.resolve=function(t){if(t instanceof e)return t;if(null===t)return u;if(void 0===t)return c;if(!0===t)return o;if(!1===t)return i;if(0===t)return f;if(""===t)return a;if("object"===typeof t||"function"===typeof t)try{var n=t.then;if("function"===typeof n)return new e(n.bind(t))}catch(r){return new e(function(t,n){n(r)})}return s(t)},e.all=function(t){var n=Array.prototype.slice.call(t);return new e(function(t,r){if(0===n.length)return t([]);var o=n.length;function i(u,c){if(c&&("object"===typeof c||"function"===typeof c)){if(c instanceof e&&c.then===e.prototype.then){for(;3===c._i;)c=c._j;return 1===c._i?i(u,c._j):(2===c._i&&r(c._j),void c.then(function(t){i(u,t)},r))}var f=c.then;if("function"===typeof f)return void new e(f.bind(c)).then(function(t){i(u,t)},r)}n[u]=c,0===--o&&t(n)}for(var u=0;u<n.length;u++)i(u,n[u])})},e.reject=function(t){return new e(function(n,r){r(t)})},e.race=function(t){return new e(function(n,r){t.forEach(function(t){e.resolve(t).then(n,r)})})},e.prototype.catch=function(t){return this.then(null,t)}},78:function(t,n,r){var e=r(16),o=r(20),i=r(31),u=r(49),c=r(137),f=function t(n,r,f){var a,s,l,p,v=n&t.F,y=n&t.G,h=n&t.P,d=n&t.B,b=y?e:n&t.S?e[r]||(e[r]={}):(e[r]||{}).prototype,g=y?o:o[r]||(o[r]={}),m=g.prototype||(g.prototype={});for(a in y&&(f=r),f)l=((s=!v&&b&&void 0!==b[a])?b:f)[a],p=d&&s?c(l,e):h&&"function"==typeof l?c(Function.call,l):l,b&&u(b,a,l,n&t.U),g[a]!=l&&i(g,a,p),h&&m[a]!=l&&(m[a]=l)};e.core=o,f.F=1,f.G=2,f.S=4,f.P=8,f.B=16,f.W=32,f.U=64,f.R=128,t.exports=f},79:function(t,n,r){var e=r(33);t.exports=function(t,n){if(!e(t))return t;var r,o;if(n&&"function"==typeof(r=t.toString)&&!e(o=r.call(t)))return o;if("function"==typeof(r=t.valueOf)&&!e(o=r.call(t)))return o;if(!n&&"function"==typeof(r=t.toString)&&!e(o=r.call(t)))return o;throw TypeError("Can't convert object to primitive value")}},80:function(t,n,r){var e=r(21).f,o=r(19),i=r(12)("toStringTag");t.exports=function(t,n,r){t&&!o(t=r?t:t.prototype,i)&&e(t,i,{configurable:!0,value:n})}},81:function(t,n,r){var e=r(139),o=r(86);t.exports=Object.keys||function(t){return e(t,o)}},82:function(t,n){var r={}.toString;t.exports=function(t){return r.call(t).slice(8,-1)}},83:function(t,n){t.exports=function(t){if(void 0==t)throw TypeError("Can't call method on  "+t);return t}},84:function(t,n){var r=Math.ceil,e=Math.floor;t.exports=function(t){return isNaN(t=+t)?0:(t>0?e:r)(t)}},85:function(t,n,r){var e=r(50)("keys"),o=r(35);t.exports=function(t){return e[t]||(e[t]=o(t))}},86:function(t,n){t.exports="constructor,hasOwnProperty,isPrototypeOf,propertyIsEnumerable,toLocaleString,toString,valueOf".split(",")},87:function(t,n){n.f={}.propertyIsEnumerable},88:function(t,n,r){var e=r(83);t.exports=function(t){return Object(e(t))}},89:function(t,n){t.exports={}}},[[525,0,1]]]);
//# sourceMappingURL=polyfill-03a8e50b2d.chunk.js.map