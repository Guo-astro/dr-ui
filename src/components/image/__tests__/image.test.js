import React from 'react';
import renderer from 'react-test-renderer';
import { testCases } from './image-test-cases.js';

describe('image', () => {
  describe(testCases.basic.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.noDefaults.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.noDefaults;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
