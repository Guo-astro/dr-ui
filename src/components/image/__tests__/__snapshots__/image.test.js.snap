// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image Basic renders as expected 1`] = `
<div
  className="py24 px24 bg-gray-faint  round"
>
  <div />
  <div
    className="mt12"
    style={
      Object {
        "color": "#557292",
      }
    }
  >
    We are having eggs for dinner
  </div>
</div>
`;

exports[`image turn off defaults renders as expected 1`] = `
<div
  className=""
>
  <div />
</div>
`;
